src/common/thread_measurement.cpp

```diff
@@ -21,20 +20,48 @@ void StartMeasurementTask(void const *argument) {
 
     uint32_t next_fs_cycle = HAL_GetTick();
     uint32_t next_sg_cycle = HAL_GetTick();
+    uint32_t next_fs_autoload_cycle = HAL_GetTick();
+
+    static bool old_fs_state = false;
+    static bool run_first = true;
 
     for (;;) {
         marlin_client_loop();
         uint32_t now = HAL_GetTick();
 
         // sample filament sensor
-        if (checkTimestampsAscendingOrder(next_fs_cycle, now)) {
+        if (next_fs_cycle <= now) {
             fs_cycle();
             // call fs_cycle every ~50 ms
             next_fs_cycle = now + 50;
         }
 
+        // sample filament autoload
+        // call every ~100ms
+        marlin_update_vars(MARLIN_VAR_MSK_FS);
+        marlin_vars_t *vars = marlin_update_vars(MARLIN_VAR_SD_PRINT);
```

Comment:
/question/ I am not really familiar with marlin_client's api, but shouldn't it be possible to read this in one go?

```C
marlin_vars_t *vars = marlin_update_vars(MARLIN_VAR_SD_PRINT | MARLIN_VAR_MSK_FS);
```

Reply:
/question/ Why don't we check for the `next_fs_autoload_cycle` first and then check whether autoload is enabled? Updating marlin vars seems like more expensive operation we might not want to do on every cycle here.

Reply:
Solved.